app/models/orchestration_stack.rb

```diff
@@ -139,4 +139,34 @@ def raw_exists?
   rescue MiqException::MiqOrchestrationStackNotExistError
     false
   end
+
+  def refresh_ems
+    self.class.refresh_ems(ext_management_system.id, ems_ref)
+  end
+
+  def self.refresh_ems(manager_id, manager_ref)
+    manager = ExtManagementSystem.find_by(:id => manager_id)
+
+    unless manager
+      raise _("No %{table} defined") % {:table => ui_lookup(:table => "ext_management_systems")}
+    end
+    unless manager.has_credentials?
+      raise _("No %{table} credentials defined") % {:table => ui_lookup(:table => "ext_management_systems")}
+    end
+    unless manager.authentication_status_ok?
+      raise _("%{table} failed last authentication check") % {:table => ui_lookup(:table => "ext_management_systems")}
+    end
```

Comment:
We should move these conditions to a common method, the same code pattern is in `ext_management_system`, `host`, and two places in `vm_or_template`

Reply:
yes, I plan to extract it